src/query/service/src/pipelines/processors/transforms/window/transform_window.rs

```diff
@@ -652,6 +647,58 @@ impl<T: Number> TransformWindow<T> {
         }
         false
     }
+
+    #[inline]
+    fn find_nth_non_null_value(
+        &self,
+        mut cur: RowPtr,
+        arg: usize,
+        ignore_null: bool,
+        advance: bool,
+    ) -> Scalar {
+        let block = &self.blocks.get(cur.block - self.first_block).unwrap().block;
+        let mut col = block.get_by_offset(arg).to_column(block.num_rows());
```

Comment:
The return value is a `Scalar`, we can deal with `Scalar` argument separately, without having to convert it to a column.